### PR TITLE
Use configured Elevate payment gateway to override default

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
Use configured Elevate payment gateway to override default.
# Critical Changes

# Changes
- We've added the ability to choose an Elevate payment gateway on a Gift Entry template to override the default gateway.

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
Track [W-11565070](https://gus.lightning.force.com/lightning/r/a07EE000013ZJ2lYAG/view)

**Comment by npsp-reedestockton** [comment]
@voduyemi @daniel-fuller Tagged you for review on this work item. Thanks!

**Comment by npsp-reedestockton** [comment]
@lparrott Added you as a reviewer on this PR. Thanks!

File: force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js [hunk 1]
```diff
@@ -18,21 +18,28 @@ import psShowPaymentMethods from '@salesforce/label/c.psShowPaymentMethods';
...
 import RD2_Credit_Card_Payment_Method_Label from '@salesforce/label/c.RD2_Credit_Card_Payment_Method_Label';
 import GeGatewaySettings from 'c/geGatewaySettings';
-import { fireEvent } from 'c/pubsubNoPageRef';
+import { fireEvent, registerListener } from 'c/pubsubNoPageRef';
 import { isEmpty, showToast } from 'c/utilCommon';
 
+const TEMPLATE_CSS_CLASS = 'template-top';
+const GIFT_ENTRY_CSS_CLASS = 'gift-entry-top';
+const TOKENIZE_ID = 'TOKENIZE';
+const MANAGEMENT_ID = 'MANAGEMENT';
```
Comment (lparrott) [inline force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js:27]: Are these constants used to indicate which page this change is happening on? If so, can we update the variable name and or string to indicate that? Even `parentId` feels like it should be either a record ID or Elevate ID, not a constant string value.

**Review by lparrott — COMMENTED** [review]
First pass looks good, a bit confusing with all of the overlap between this and #7112

File: force-app/main/default/lwc/geGatewaySettings/geGatewaySettings.js [hunk 2]
```diff
@@ -1,20 +1,52 @@
...
         registerListener('updateElevateSettings', this.settings, this);
     }
 
+    initDecryptedElevateSettings(elevateSettings) {
+        this.isGiftEntryBatch = true;
+        this.elevateSettings = elevateSettings;
+        if (this.elevateSettings?.uniqueKey) {
+            this.decryptElevateGateway().then(error => {
+                if (this.elevateSettings?.uniqueKey && !this.decryptedGatewayId) {
+                    showToast(psGatewayNotValid, details, 'error', 'sticky');
```
Comment (voduyemi) [inline force-app/main/default/lwc/geGatewaySettings/geGatewaySettings.js:26]: Where are we getting this `details` property from? Also we should be using the `catch` method to catch the potential error from this promise

File: force-app/main/default/lwc/gePaymentGatewayManagement/gePaymentGatewayManagement.js [hunk 3]
```diff
@@ -36,9 +36,11 @@ import insufficientPermissions from '@salesforce/label/c.commonInsufficientPermi
 import commonAdminPermissionErrorMessage from '@salesforce/label/c.commonAdminPermissionErrorMessage';
 
 import setGatewayId from '@salesforce/apex/PS_GatewayManagement.setGatewayId';
```
Comment (voduyemi) [inline force-app/main/default/lwc/gePaymentGatewayManagement/gePaymentGatewayManagement.js:38]: A general comment on this JS class, how are we planning to integrate @screcco-sfdo changes here?

File: force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js [hunk 4]
```diff
@@ -280,13 +332,38 @@ export default class GeGatewaySelectWidget extends LightningElement {
...
+    }
+
     async updateElevateSettings() {
-        let elevateSettings = {
-            uniqueKey: this.isGatewayAssignmentEnabled ?
-                await encryptGatewayId( {gatewayId: this.selectedGateway}) : null,
-            isACHEnabled: this.isACHEnabled,
-            isCreditCardEnabled: this.isCreditCardEnabled
+        if (this._widgetMode !== GATEWAY_MANAGEMENT_MODE) {
+            let elevateSettings = {
```
Comment (voduyemi) [inline force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js:345]: This variable should be declares as `const`

File: force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js [hunk 5]
```diff
@@ -77,24 +100,46 @@ export default class GeGatewaySelectWidget extends LightningElement {
     async connectedCallback() {
         await this.init();
 
-        if (this.isGatewayAssignmentEnabled) {
+        if (this._widgetMode !== PAYMENT_METHOD_MODE) {
```
Comment (voduyemi) [inline force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js:103]: We should be calling `shouldShowGateways` here

File: force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js [hunk 6]
```diff
@@ -57,14 +62,32 @@ export default class GeGatewaySelectWidget extends LightningElement {
     _firstDisplay = true;
     _defaultTemplateId = null;
     _defaultGatewayId = null;
+    _widgetMode;
 
     async init() {
-        let gatewayAssignmentSettings = JSON.parse(await getGatewayAssignmentSettings());
+        try {
+            let gatewayAssignmentSettings = JSON.parse(await getGatewayAssignmentSettings());
```
Comment (voduyemi) [inline force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js:69]: This variable should be declared as `const`

**Review by voduyemi — CHANGES_REQUESTED** [review]

File: force-app/main/default/lwc/geGatewaySettings/geGatewaySettings.js (hunk 2 above)
Comment (npsp-reedestockton) [reply force-app/main/default/lwc/geGatewaySettings/geGatewaySettings.js:26]: details is passed from handleErrors.

Comment (npsp-reedestockton) [reply force-app/main/default/lwc/geGatewaySettings/geGatewaySettings.js:26]: I'm wrong. I'll take care of that. Copy/paste error...

File: force-app/main/default/lwc/gePaymentGatewayManagement/gePaymentGatewayManagement.js (hunk 3 above)
Comment (npsp-reedestockton) [reply force-app/main/default/lwc/gePaymentGatewayManagement/gePaymentGatewayManagement.js:38]: As soon as his PR is merged, I will integrate the two.

**Comment by lparrott** [comment]
Merged as part of #7112

**Pull request closed without merging**